pkg/models/event.go

```diff
@@ -57,7 +57,12 @@ func (e *Event) WithMessage(message string) *Event {
 
 // WithError returns a new Event with the given error.
 func (e *Event) WithError(err error) *Event {
-	e.Message = err.Error()
+	return e.WithErrorMessage(err.Error())
+}
+
+// WithErrorMessage returns a new Event with the given error.
+func (e *Event) WithErrorMessage(message string) *Event {
+	e.Message = message
 	return e.WithDetail(DetailsKeyIsError, "true")
 }
```

Comment:
Not enough. `WithMessage` accepts a string and doesn't' flag the event as `DetailsKeyIsError`. `WithError` accepts an error type. The alternatives of not adding `WithErrorMessage` are:
- Changing `WithError` to accept a string instead of error type. This doesn't sound right
- Construct an error type of a string to pass it to `WithError`. Doesn't sound right for callers to construct errors only to pass them to the event builder and not return the error!
- Calling `WithError` and then manually setting `DetailsKeyIsError` in the caller. We will have more similar cases and I prefer to simplify the creation of events and have one place for common setup of events

Reply:
My concern is DevX as this change obscures usages of this package.  Pass WithError an error as I suggested below. We are returning the event that encapsulates the error to the caller so I'll good there.